src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -151,3 +152,67 @@ def signature_hash_algorithm(self):
             raise UnsupportedAlgorithm(
                 "Signature algorithm OID:{0} not recognized".format(oid)
             )
+
+    @property
+    def extensions(self):
+        # TODO: remove this when we finish landing support for extensions.
+        warnings.warn(
+            "Extension support is not fully implemented. Unsupported "
+            "extensions are IGNORED at this time."
+        )
+        extensions = []
+        extcount = self._backend._lib.X509_get_ext_count(self._x509)
+        if extcount > 0:
+            bc = self._get_extension_by_nid(
+                self._backend._lib.NID_basic_constraints, "basicConstraints"
+            )
+            if bc:
+                ext = self._build_basic_constraints(bc)
+                extensions.append(ext)
+
+        return x509.Extensions(extensions)
+
+    def _get_extension_by_nid(self, nid, name):
+        index = self._backend._lib.X509_get_ext_by_NID(self._x509, nid, -1)
+        if index >= 0:
+            if self._backend._lib.X509_get_ext_by_NID(
+                self._x509, nid, index
+            ) > 0:
+                raise x509.DuplicateExtension(
+                    "Duplicate {0} extension found".format(name)
+                )
+
+            # This does not retain the value so we don't need to free it.
+            ext = self._backend._lib.X509_get_ext(self._x509, index)
+            assert ext != self._backend._ffi.NULL
+            return ext
+        else:
+            return None
+
+    def _build_basic_constraints(self, ext):
+        bc_st = self._backend._lib.X509V3_EXT_d2i(ext)
+        assert bc_st != self._backend._ffi.NULL
+        bc = self._backend._ffi.cast("BASIC_CONSTRAINTS *", bc_st)
+        bc = self._backend._ffi.gc(
+            bc, self._backend._lib.BASIC_CONSTRAINTS_free
+        )
+        # The byte representation of an ASN.1 boolean true is \xff. OpenSSL
+        # chooses to just map this to its ordinal value, so true is 255 and
+        # false is 0.
+        ca = bc.ca == 255
+        if bc.pathlen == self._backend._ffi.NULL:
+            path_length = None
+        else:
+            bn = self._backend._lib.ASN1_INTEGER_to_BN(
+                bc.pathlen, self._backend._ffi.NULL
+            )
+            assert bn != self._backend._ffi.NULL
+            bn = self._backend._ffi.gc(bn, self._backend._lib.BN_free)
+            path_length = self._backend._bn_to_int(bn)
+
+        bc = x509.BasicConstraints(ca, path_length)
+
+        crit = self._backend._lib.X509_EXTENSION_get_critical(ext)
+        assert crit != self._backend._ffi.NULL
+        critical = crit == 1
+        return x509.Extension(x509.OID_BASIC_CONSTRAINTS, critical, bc)
```

Comment:
Creating the `x509.Extension` object should happen outside of this method, since it's the same in every extension.